python/packages/ai/teams/ai/planners/planner.py

```diff
@@ -0,0 +1,41 @@
+"""
+Copyright (c) Microsoft Corporation. All rights reserved.
+Licensed under the MIT License.
+"""
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+
+from botbuilder.core import TurnContext
+
+from ...state import TurnState
+from .plan import Plan
+
+
+class Planner(ABC):
+    """
+    A planner is responsible for generating a plan that the AI system will execute.
+    """
+
+    @abstractmethod
+    async def begin_task(self, context: TurnContext, state: TurnState) -> Plan:
```

Comment:
also I noticed that even those the JS is passing in that class it is not being references/used at all

Reply:
I see, maybe we should remove it from the method signature for consistency

Reply:
ya we can create a ticket to refactor in the other languages, in general its bad practice for a parent object to pass a reference if itself into a child class because it causes cyclic dependency problems, python doesn't handle that very well